daffodil-io/src/main/scala/edu/illinois/ncsa/daffodil/io/DataStreamCommonImplMixin.scala

```diff
@@ -35,14 +35,29 @@ import edu.illinois.ncsa.daffodil.util.MaybeChar
 import edu.illinois.ncsa.daffodil.util.Logging
 import edu.illinois.ncsa.daffodil.api.DataStreamLimits
 import edu.illinois.ncsa.daffodil.schema.annotation.props.gen.BitOrder
+import edu.illinois.ncsa.daffodil.util.Maybe
+import edu.illinois.ncsa.daffodil.util.Maybe._
+import edu.illinois.ncsa.daffodil.exceptions.Assert
 
 trait DataStreamCommonState {
 
-  //  var binaryFloatRep: BinaryFloatRep = BinaryFloatRep.Ieee
-  var priorBitOrder: BitOrder = BitOrder.MostSignificantBitFirst
-  //  var maybeCharWidthInBits: MaybeInt = MaybeInt.Nope
-  //  var encodingMandatoryAlignmentInBits: Int = 8
-  //  var maybeUTF16Width: Maybe[UTF16Width] = Maybe(UTF16Width.Fixed)
+  /**
+   * Keeps track of the bitOrder of the last operation on the
+   * data stream.
+   */
+  private var maybePriorBitOrder_ : Maybe[BitOrder] = Maybe(BitOrder.MostSignificantBitFirst)
```

Comment:
I changed this to MSBF because there was a case where we need to return a bitOrder and no setup has been done. I should change this back to Nope and see what breaks exactly and see. Because clearly the right thing is for this to be Nope initially.